src/hotspot/share/gc/shenandoah/shenandoahDegeneratedGC.cpp

```diff
@@ -164,15 +164,15 @@ void ShenandoahDegenGC::op_degenerated() {
           // we will rescan the roots on this safepoint.
           heap->old_generation()->transfer_pointers_from_satb();
         }
-      }
 
-      if (_degen_point == ShenandoahDegenPoint::_degenerated_roots) {
-        // We only need this if the concurrent cycle has already swapped the card tables.
-        // Marking will use the 'read' table, but interesting pointers may have been
-        // recorded in the 'write' table in the time between the cancelled concurrent cycle
-        // and this degenerated cycle. These pointers need to be included the 'read' table
-        // used to scan the remembered set during the STW mark which follows here.
-        _generation->merge_write_table();
```

Comment:
Interesting. I am trying to understand this diff (the change in nesting looks curious because I didn't see anything change above. As a result, this specifically generational work(?) may have been pulled into the `else {` arm of the if-then-else at lines 145-149, introducing a bug?

Or is this intentional and I am missing some subtle reasoning here?

Reply:
Or may be just an issue with how this displays in the diffs. Worth a quick check.

Reply:
Good eye. I pulled this into the preceding `else` block because `merge_write_table` really means `merge_write_card_table`. It's only for the generational mode, so we don't need to execute it in non-generational modes as was done before this change.

Reply:
ah, ok. I see that now. The diff had looked weird to me. Looks good with yr help above :-)